specification/_types/Errors.ts

```diff
@@ -36,7 +36,7 @@ export class ErrorCause
   /**
    * A human-readable explanation of the error, in english
    */
-  reason: string
```

Comment:
Hmm! When can an error not have a `reason`?

Reply:
I went digging to understand this, as it was surprising but happening in the wild ;-)

`ErrorCause` is the [JSON representation of a Java exception](https://github.com/elastic/elasticsearch/blob/e03c2b0161e31f72dd848000efacab97f8c2cdbd/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L523) (a `Throwable`). The `reason` field is the result of Java's `Throwable.getMessage()`, which [according to the docs](https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#getMessage--) can be `null`. In practice this never (I guess) happens for the top-level error, which is generally an `ElasticsearchException`. The error here happened in a deeper root cause, where apparently `getMessage()` returned `null` 🤷

Reply:
null can happen when it's same exception as before in case of `NullPointerExceptions` - jvm can optimize them out